src/assets/scripts/client/UiController.js

```diff
@@ -293,95 +319,133 @@ export default class UiController {
+        this._buildAirportList();
+        this._drawAirportListFooter();
+    }
+
+    /**
+     * @for UiController
+     * @method _buildAirportList
+     * @private
+     */
+    _buildAirportList() {
+        // clear out the contents of this element
+        // this method will run every time a user changes the `INCLUDE_WIP_AIPRORTS` option
+        this.$airportList.empty();
+
+        const airports = _keys(this._airportController.airport.airports).sort();
+        const shouldShowWipAirports = this._gameController.getGameOption(GAME_OPTION_NAMES.INCLUDE_WIP_AIRPORTS) === 'yes';
         let difficulty = '';
-        let airport;
 
         for (let i = 0; i < airports.length; i++) {
-            airport = prop.airport.airports[airports[i]];
-
-            switch (airport.level) {
-                case 'beginner':
-                    difficulty = icon;
-                    break;
-                case 'easy':
-                    difficulty = icon.repeat(2);
-                    break;
-                case 'medium':
-                    difficulty = icon.repeat(3);
-                    break;
-                case 'hard':
-                    difficulty = icon.repeat(4);
-                    break;
-                case 'expert':
-                    difficulty = icon.repeat(5);
-                    break;
-                default:
-                    difficulty = '?';
-                    break;
+            const { name, icao, level, wip } = this._airportController.airport.airports[airports[i]];
+
+            if (!shouldShowWipAirports && wip) {
+                continue;
             }
 
-            // TODO: move to a template const
-            const { name, icao, wip } = airport;
-            const flagIcon = (wip === true) ? ' &#9983' : '';
+            difficulty = this._buildAirportListIconForDifficultyLevel(level);
+            const flagIcon = wip
+                ? ' &#9983'
+                : '';
             const $airportListItem = $(this.buildAirportListItemTemplate(icao, difficulty, name, flagIcon));
 
             // TODO: replace with an onClick() handler
-            $airportListItem.click(airport.icao.toLowerCase(), (event) => {
-                if (event.data !== window.airportController.airport_get().icao) {
-                    window.airportController.airport_set(event.data);
+            $airportListItem.click(icao.toLowerCase(), (event) => {
+                if (event.data !== this._airportController.airport_get().icao) {
+                    this._airportController.airport_set(event.data);
                     this.ui_airport_close();
                 }
             });
 
             this.$airportList.append($airportListItem);
         }
+    }
+
+    /**
+     *  @for UiController
+     * @method _buildAirportListIconForDifficultyLevel
+     * @param difficultyLevel {string}
+     * @return difficulty {string}
+     * @private
+     */
+    _buildAirportListIconForDifficultyLevel(difficultyLevel) {
```

Comment:
some descriptions with the docblocks, please. And there's an extra space before the `@for`.